### PR TITLE
Align package ID validation with the client

### PR DESCRIPTION
**Pull request by scottbommarito**
We now use `NuGet.Packaging`'s implementation of the `PackageIdValidator` instead of our own class. Additionally, we now use a constant for max package ID length everywhere instead of `128`.

- [x] use the new version of `NuGet.Packaging` with the `PackageIdValidator.MaxPackageIdLength` constant exposed (currently waiting on a build)

### CODE REVIEW

**Comment by scottbommarito** [comment]
https://github.com/NuGet/NuGetGallery/issues/3142

File: src/NuGetGallery/Views/Users/Profiles.cshtml [hunk 1]
```diff
@@ -9,7 +10,7 @@
     @if (!String.IsNullOrEmpty(Model.EmailAddress))
     {
         <p>
-            @ViewHelpers.GravatarImage(Model.EmailAddress, Model.Username, 128)
```
Comment (skofman1) [inline src/NuGetGallery/Views/Users/Profiles.cshtml]: >128 [](start = 75, length = 3)

is this related to package id length? Seems like user information

**Comment by skofman1** [comment]
1 comment. Otherwise looks good.

File: src/NuGetGallery/Services/PackageService.cs [hunk 2]
```diff
@@ -612,10 +612,10 @@ private static void ValidateNuGetPackageMetadata(PackageMetadata packageMetadata
 
                 foreach (var dependency in packageDependencies.SelectMany(s => s.Packages))
                 {
-                    // NuGet.Core compatibility - dependency package id can not be > 128 characters
-                    if (dependency.Id != null && dependency.Id.Length > CoreConstants.MaxPackageIdLength)
+                    // NuGet.Core compatibility - dependency package id can not be > PackageIdValidator.MaxPackageIdLength characters
```
Comment (shishirx34) [inline src/NuGetGallery/Services/PackageService.cs]: nit: `cannot`

not particularly a fan of putting variable in the comments.

**Comment by scottbommarito** [comment]
Waiting for a new build of `NuGet.Client` with the required `MaxPackageIdLength` field to reference.

**Comment by dtivel** [comment]
Looks good, but I'm curious...

The previous ID length limit was 128, while the new length limit is 100.  Are there currently any packages with an ID length greater than 100?  If yes, how will this change affect package owners uploading a newer version of said package?

**Comment by scottbommarito** [comment]
@dtivel As far as I understand, `NuGet.exe` has and has always had a 100 character limit, so unless somebody had manually created a `.nupkg` with a long ID, it should be impossible to create a package of that length and then upload it to our service. Just to be sure, however, I just verified that we do not have any packages with an ID longer than 100 on the gallery..

**Comment by dtivel** [comment]
Thanks, Scott.

**Comment by scottbommarito** [comment]
closing because this was accidentally picked to merge into master...
